libraries/bot-builder/src/test/java/com/microsoft/bot/builder/teams/TeamsInfoTests.java

```diff
@@ -81,6 +85,52 @@ public void TestSendMessageToTeamsChannel() {
         handler.onTurn(turnContext).join();
     }
 
+    @Test
+    public void TestSendMessageToTeamsChannelWithCloudAdapterSupport() {
+        String expectedTeamsChannelId = "teams-channel-id";
+        String expectedAppId = "app-id";
+        String expectedServiceUrl = "service-url";
+        String expectedActivityId = "activity-id";
+        String expectedConversationId = "conversation-id";
+
+        Activity requestActivity = new Activity(ActivityTypes.MESSAGE);
+        requestActivity.setServiceUrl(expectedServiceUrl);
+
+        TestCreateConversationAdapter adapter = new TestCreateConversationAdapter(expectedActivityId, expectedConversationId);
+
+        TurnContext turnContextMock = Mockito.mock(TurnContext.class);
+        Mockito.when(turnContextMock.getActivity()).thenReturn(requestActivity);
+        Mockito.when(turnContextMock.getAdapter()).thenReturn(adapter);
+
+        Activity activity = new Activity(ActivityTypes.MESSAGE);
+        activity.setText("Test-SendMessageToTeamsChannelAsync");
+        activity.setChannelId(Channels.MSTEAMS);
+
+        TeamsChannelData data = new TeamsChannelData();
+
+        TeamInfo teamInfo = new TeamInfo();
+        teamInfo.setId("team-id");
+
+        data.setTeam(teamInfo);
+
+        activity.setChannelData(data);
+
+        Pair<ConversationReference, String> r = TeamsInfo.sendMessageToTeamsChannel(turnContextMock, activity, expectedTeamsChannelId, expectedAppId).join();
+
+        Assert.assertEquals(expectedConversationId, r.getLeft().getConversation().getId());
+        Assert.assertEquals(expectedActivityId, r.getRight());
+        Assert.assertEquals(expectedAppId, adapter.getAppId());
+        Assert.assertEquals(Channels.MSTEAMS, adapter.getChannelId());
+        Assert.assertEquals(expectedServiceUrl, adapter.getServiceUrl());
+        Assert.assertNull(adapter.getAudience());
```

Comment:
Maybe I'm missing something, but aren't this block's assertions validating that the object we just created above have the value we also set a few lines above?

Reply:
Dropping this as it's ported 1-1 from the C# PR